pkg/controllers/clusteroperator_controller.go

```diff
@@ -251,7 +256,48 @@ func (r *CloudOperatorReconciler) provisioningAllowed(ctx context.Context, infra
 		return false, nil
 	}
 
-	// TODO: Add part confirming migration is finished for KCM before allowing provisioning
+	ownedByKCM, err := r.isCloudControllerOwnedByKCM(ctx)
+	if err != nil {
+		return false, err
+	}
+	if ownedByKCM {
+		// KCM resource found and it owns Cloud provider
+		klog.Infof("Kube Controller Manager still owns Cloud Controller. Skipping...")
+
+		if err := r.setStatusAvailable(ctx); err != nil {
+			klog.Errorf("Unable to sync cluster operator status: %s", err)
+			return false, err
+		}
+
+		return false, nil
+	}
 
 	return true, nil
 }
+
+func (r *CloudOperatorReconciler) isCloudControllerOwnedByKCM(ctx context.Context) (bool, error) {
+	kcm := &operatorv1.KubeControllerManager{}
+	err := r.Get(ctx, client.ObjectKey{Name: kcmResourceName}, kcm)
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
What happens when the object doesn't exist or it hasn't set the condition yet (because it's not running?)